File: moon_img_compress/src/main/java/com/mljr/moon/imgcompress/util/Preconditions.java
package com.mljr.moon.imgcompress.util;

import android.support.annotation.Nullable;

/**
 * Author： fanyafeng
 * Date： 17/12/27 上午10:51
 * Email: fanyafeng@live.cn
 */
public final class Preconditions {

    private Preconditions(){}

    public static <T> T checkNotNull(T reference) {
        if (reference == null) {
            throw new NullPointerException();
        }
        return reference;
    }

    public static <T> T checkNotNull(T reference, @Nullable Object errorMessage) {
        if (reference == null) {
            throw new NullPointerException(String.valueOf(errorMessage));
        }
        return reference;
    }
}

File: ripple_dialog/src/main/java/com/ripple/dialog/util/Preconditions.kt
package com.ripple.dialog.util

/**
 * Author： fanyafeng
 * Date： 2019-12-27 16:26
 * Email: fanyafeng@live.cn
 */
internal object Preconditions {
    @JvmStatic
    fun <T> checkNotNull(reference: T?, errorMessage: Any): T {
        if (reference == null) {
            throw NullPointerException(errorMessage.toString())
        }
        return reference
    }

}